scripts/test_translator.py

```diff
@@ -63,7 +63,7 @@ def __init__(self, path: str, link_name: str,
 
 
 class CFile:
-    def __init__(self, log_level: str, path: str, flags: Set[str] = None) -> None:
+    def __init__(self, log_level: str, path: str, flags: Set[str] = set()) -> None:
         if not flags:
             flags = set()
```

Comment:
Yeah, the initial code does it correctly except for the types.  It's annoying that `mypy` can't refine an existing variable's type, though, and argument names are public (for keyword args).

Instead, I think it's better to restrict the `flags` type to what's needed, `Container`, with its `__contains__` method and `in` operator.  Thus, we can set the default value to `()`, which is immutable.

```suggestion
    def __init__(self, log_level: str, path: str, flags: Container[str] = ()) -> None:
```

Reply:
Resolved as we don't mutate this set either.